src/main.ts

```diff
@@ -39,142 +64,237 @@ async function processIssues(
   const issues = await client.issues.listForRepo({
     owner: github.context.repo.owner,
     repo: github.context.repo.repo,
-    state: 'open',
+    state: "open",
     per_page: 100,
-    page: page
-  });
+    page: page,
+  })
 
-  operationsLeft -= 1;
+  operationsLeft -= 1
 
   if (issues.data.length === 0 || operationsLeft === 0) {
-    return operationsLeft;
+    return operationsLeft
   }
 
-  for (var issue of issues.data.values()) {
-    core.debug(`found issue: ${issue.title} last updated ${issue.updated_at}`);
-    let isPr = !!issue.pull_request;
+  for (const issue of issues.data.values()) {
+    core.debug(
+      `found issue: "${issue.title}" (#${issue.number}) ― last updated: ${issue.updated_at}`
+    )
+    let isPr = !!issue.pull_request
 
-    let staleMessage = isPr ? args.stalePrMessage : args.staleIssueMessage;
+    // If no stale message for issue/PR is defined, skip it
+    let staleMessage = isPr ? args.STALE_PR_MESSAGE : args.STALE_ISSUE_MESSAGE
     if (!staleMessage) {
-      core.debug(`skipping ${isPr ? 'pr' : 'issue'} due to empty message`);
-      continue;
+      core.debug(
+        `skipping ${
+          isPr ? "pr" : "issue"
+        } due to empty stale message in the action's config`
+      )
+      continue
     }
 
-    let staleLabel = isPr ? args.stalePrLabel : args.staleIssueLabel;
-    let exemptLabel = isPr ? args.exemptPrLabel : args.exemptIssueLabel;
+    // If the issue has at least one assignee, skip it
+    if (issue.assignee !== null && issue.assignees.length >= 1) {
+      core.debug(
+        `skipping ${
+          isPr ? "pr" : "issue"
+        } because at least one person is assigned`
+      )
+      continue
+    }
 
-    if (exemptLabel && isLabeled(issue, exemptLabel)) {
-      continue;
+    let staleLabel = isPr ? args.STALE_PR_LABEL : args.STALE_ISSUE_LABEL
+    let exemptLabels = isPr ? args.EXEMPT_PR_LABELS : args.EXEMPT_ISSUE_LABELS
+
+    // Check if at least one exempt label is present
+    if (exemptLabels.some(exemptLabel => isLabeled(issue, exemptLabel))) {
+      continue
     } else if (isLabeled(issue, staleLabel)) {
-      if (wasLastUpdatedBefore(issue, args.daysBeforeClose)) {
-        operationsLeft -= await closeIssue(client, issue);
+      // Check if the staleLabel is present & the last update is longer than the close days. If yes, close the issue
+      if (wasLastUpdatedBefore(issue, args.DAYS_BEFORE_CLOSE)) {
+        operationsLeft -= await closeIssue(
+          client,
+          issue,
+          args.CLOSE_MESSAGE,
+          args.DRY_RUN
+        )
+        // If a user commented since the addition of the stale label, remove the label
+      } else if (
+        await issueHasActivity(client, issue, args.DAYS_BEFORE_STALE)
```

Comment:
This doesn't exactly check if user commented on issue since addition of label (`DAYS_BEFORE_STALE` is used as we don't have date/time of applying stale label).

I need to think if this is a problem or not (might depend on DAYS_BEFORE_CLOSE and DAYS_BEFORE_STALE configuration), but comment above might need to be updated